core/platform/users/gae_current_user_services.py

```diff
@@ -32,8 +32,14 @@ def create_login_url(slug):
 
 
 def create_logout_url(slug):
-    """Creates a logout url."""
-    logout_url = utils.set_url_query_parameter('/logout', 'return_url', slug)
+    """Creates a logout url.
+    Logout from exploration editor is handled separately."""
+    if 'create' in slug.split('/'):
```

Comment:
This isn't the right place to do this -- the platform layer shouldn't know about 'create'. It needs to be done somewhere higher up, perhaps in the editor page controller.

Reply:
i didn't got a way to change logout href without modifing 'create_logout_url()'.

the other way could be to handle exp_editor logout in LogoutPage(as earlier) and check if it's from exp_editor, but this would require importing exp_services in base.py.

i think this is fine, instead of creating a new handler, what are your thought on this.

Reply:
base.py shouldn't import exp_services directly, or even know about the different types of pages. I suppose one way around this is to create a get_logout_url() method in the base controller class and override it in the editor page controller class.